src/utils/index.ts

```diff
@@ -317,3 +324,38 @@ export function maybeParse(data: Quad[] | string): Quad[] {
         return data;
     }
 }
+
+export function findRecordTimestamp(
```

Comment:
Hm, this implementation doesn't fit with the implementation.
Either you should make the lastMemberTimestamp optional, or the name should be changed to indicate that this function checks the ordering.
Also, in my opinion out of ordered things should always be allowed, but within a buffer window, what do you think?